src/toast/Toast.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import "@clr/icons/clr-icons.min.css";
+import "@webcomponents/custom-elements/custom-elements.min.js";
+import "@clr/icons/clr-icons.min.js";
+import MessageType from "./MessageTypes";
+import Direction from "./Directions";
+import {Icon} from "../icon";
+import "./styles.css";
+
+export interface ToastOptions {
+    type?: MessageType;
+    position?: Direction;
+    hideAfter?: number;
+}
+
+const TOAST_CLASSES = {
+    VISIBLE: "toast-visible",
+    INVISIBLE: "toast-invisible",
+    LEFT: "toast-left",
+    RIGHT: "toast-right",
+    TOP: "toast-top",
+    BOTTOM: "toast-bottom",
+    MARGIN_BOTTOM: "toast-margin-bottom",
+    MARGIN_TOP: "toast-margin-top",
+    MARGIN_RIGHT: "toast-margin-right",
+    MARGIN_LEFT: "toast-margin-left",
+    NEG_MARGIN_BOTTOM: "toast-neg-margin-bottom",
+    NEG_MARGIN_TOP: "toast-neg-margin-top",
+    NEG_MARGIN_RIGHT: "toast-neg-margin-right",
+    NEG_MARGIN_LEFT: "toast-neg-margin-left",
+};
+
+const DEFAULT_HIDE_TIME = 3;
+const MILLISEC = 1000;
+const EMPTY_MESSAGE = "";
+const PADDING_RIGHT = "padding-right";
+
+const initialState = {
+    isToastRendering: false,
```

Comment:
This is the only value we should need to keep in state.  All other values can be just properties of the Toast component.

Reply:
Updated code.